### DISCUSSION TITLE
Comparison with native node test runner

### DISCUSSION THREAD

**Discussion by philly-vanilly** [vitest-dev/vitest#4631] [General]
How does vitest compare to 'node:test'?

The [docs](https://vitest.dev/guide/comparisons.html) don't mention the new node native test runner, which seems to be the new way of doing testing https://www.sitepoint.com/node-js-20-new/

_Other runtimes took an alternative view and offered built-in tools considered essential for development. [Deno](https://deno.land/manual/basics/testing), [Bun](https://bun.sh/docs/cli/test), [Go](https://go.dev/doc/tutorial/add-a-test), and [Rust](https://doc.rust-lang.org/rust-by-example/testing.html) all offer built-in test runners. Developers have a default choice but can opt for an alternative when their project has specific requirements._

Would be nice if someone with in-depth knowledge wrote something about the advantages/disadvantages/use-cases of each

**Comment by f15u** [comment]
At the moment of writing, with the current Node LTS (v20), the test runner **does not** support `glob` patterns. This is a deal breaker for us.

We run `node` with `tsimp` (`node --import=tsimp/import`) for compiling typescript and we have `**/*.spec.ts` files.

**Reply by vorant94** [reply]
there is a new built-in [glob API](https://nodejs.org/api/fs.html#fspromisesglobpattern-options) (that is experimental, but still)

combining it with [test run API](https://nodejs.org/api/test.html#runoptions) and you can have a small wrapper script that runs built-in test runner on a glob-filtered list of files

**Reply by glokta1** [reply]
Node LTS (**v22**) supports glob pattern arguments in the test runner. See: https://nodejs.org/docs/latest-v22.x/api/test.html#running-tests-from-the-command-line

I'm running the following command to execute all tests in my project:
`node --test --require ts-node/register "**/*.test.ts"`

Note: The double quotes around the globs are important for portability as specified in the docs.

**Comment by vorant94** [comment]
maybe matter of taste or something, but for me it was tests output format (especially when some tests are failing), that made me to stick with vitest instead of built-in test runner... it just gets to much noisy as of now with all the stack traces that are thrown into stdout by built-in one
